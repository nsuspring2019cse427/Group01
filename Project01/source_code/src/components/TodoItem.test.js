import React from "react"
import Enzyme, {
    shallow
} from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import TodoItem from "./TodoItem"

describe("TodoItem Component", () => {
    beforeAll(() => {
        Enzyme.configure({
            adapter: new Adapter()
        })
    })

    it("renders correctly", () => {
        const component = shallow( < TodoItem id = {
                1
            }
            title = "Do house chores" / > )
        expect(component).toMatchSnapshot()
    })
})