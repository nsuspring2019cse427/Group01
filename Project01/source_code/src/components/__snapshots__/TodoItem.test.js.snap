exports[`TodoItem Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoItem
    id={1}
    title="Do house chores"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.label
          htmlFor={1}
        >
          <input
            id={1}
            type="checkbox"
          />
           
          Do house chores
        </styled.label>,
        <styled.button
          type="button"
        >
          Delete
        </styled.button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <input
              id={1}
              type="checkbox"
            />,
            " ",
            "Do house chores",
          ],
          "htmlFor": 1,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": 1,
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          " ",
          "Do house chores",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Delete",
          "type": "button",
        },
        "ref": null,
        "rendered": "Delete",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.label
            htmlFor={1}
          >
            <input
              id={1}
              type="checkbox"
            />
             
            Do house chores
          </styled.label>,
          <styled.button
            type="button"
          >
            Delete
          </styled.button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <input
                id={1}
                type="checkbox"
              />,
              " ",
              "Do house chores",
            ],
            "htmlFor": 1,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": 1,
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            " ",
            "Do house chores",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Delete",
            "type": "button",
          },
          "ref": null,
          "rendered": "Delete",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;